### PR TITLE
Add initial Snap support

### PR DESCRIPTION
**Pull request by diegoviola**

### CODE REVIEW

File: snap/snapcraft.yaml [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+  GWallet is a minimalistic and pragmatist crossplatform lightweight opensource
+  brainwallet for people that want to hold the most important cryptocurrencies
+  in the same application with ease and peace of mind.
+
+grade: devel # must be 'stable' to release into candidate/stable channels
+confinement: classic # use 'strict' once you have the right plugs and slots
+
+apps:
+  geewallet:
+    command: sh
```
Comment (knocte) [inline snap/snapcraft.yaml]: is specifying `sh` necessary? the launcher script has a shebang already

File: snap/snapcraft.yaml [hunk 2]
```diff
@@ -0,0 +1,42 @@
+name: geewallet # you probably want to 'snapcraft register <name>'
+base: core18 # the base snap is the execution environment for this snap
+version: '0.1' # just for humans, typically '1.2+git' or '1.3.2'
+summary: minimalistic and pragmatist crossplatform lightweight opensource brainwallet # 79 char long summary
+description: |
+  GWallet is a minimalistic and pragmatist crossplatform lightweight opensource
```
Comment (knocte) [inline snap/snapcraft.yaml]: rename from GWallet to geewallet please

Comment (diegoviola) [reply snap/snapcraft.yaml]: done

**Comment by knocte** [comment]
Good work so far! 2 things:
* Aren't you missing a mono-runtime dependency in `stage-packages`?
* Please add a script (build-snap.sh?) that creates the snap from the snapcraft.yml.

File: snap/snapcraft.yaml [hunk 3]
```diff
@@ -0,0 +1,42 @@
+name: geewallet # you probably want to 'snapcraft register <name>'
+base: core18 # the base snap is the execution environment for this snap
+version: '0.1' # just for humans, typically '1.2+git' or '1.3.2'
+summary: minimalistic and pragmatist crossplatform lightweight opensource brainwallet # 79 char long summary
```
Comment (knocte) [inline snap/snapcraft.yaml]: only 79 is allowed? damn, I guess adding "multicurrency" word is more important than "minimalistic" or "pragmatist"; which word would you take out?

**Comment by diegoviola** [comment]
> Good work so far! 2 things:
> 
>     * Aren't you missing a mono-runtime dependency in `stage-packages`?

I did that before and it turned out that mono-runtime was pulling an older release, and conflicting with the new one, so we removed the stage-packages and opted for a manual installation of the newer mono.

File: snap/snapcraft.yaml (hunk 3 above)
Comment (diegoviola) [reply snap/snapcraft.yaml]: Not sure about this.

**Comment by knocte** [comment]
> Yes, I know, and I tried this before, adding a stage-packages with mono-runtime still creates a conflict, hence why it was removed, you are welcome to try it yourself.

I see, but then, this means that for a user that hasn't installed mono on his ubuntu computer, the snap would not work when trying to install it and execute it. Right?

File: snap/snapcraft.yaml (hunk 3 above)
Comment (knocte) [reply snap/snapcraft.yaml]: yes tell me which word you dislike the most: "minimalistic" or "pragmatist"?

Comment (diegoviola) [reply snap/snapcraft.yaml]: I don't dislike either of them.

**Comment by diegoviola** [comment]
> > Yes, I know, and I tried this before, adding a stage-packages with mono-runtime still creates a conflict, hence why it was removed, you are welcome to try it yourself.
> 
> I see, but then, this means that for a user that hasn't installed mono on his ubuntu computer, the snap would not work when trying to install it and execute it. Right?

yes, it assumes that the user has mono installed system wide.

**Comment by knocte** [comment]
> yes, it assumes that the user has mono installed system wide.

Then this is an unacceptable solution, we cannot merge this like that. I'll try to find a way. Maybe if we uninstall dotnet-core, mono, and remove the apt sources after `make install`, the stage-packages get installed afterwards?

**Comment by diegoviola** [comment]
> > yes, it assumes that the user has mono installed system wide.
> 
> Then this is an unacceptable solution, we cannot merge this like that. I'll try to find a way. Maybe if we uninstall dotnet-core, mono, and remove the apt sources after `make install`, the stage-packages get installed afterwards?

I'm actually not very sure about this, I'm not a snap expert.

I suggest trying to install the snap on a clean Ubuntu installation, and check if the dependencies are also being installed system wide.

The snap will pull the dependencies for compiling the project but I don't know if those are also being installed system wide, I think they are.

I can check but I'm too exhausted right now.

**Pull request closed without merging**